label-external-pull-request/action.yml

```
@@ -0,0 +1,38 @@
+name: Label External Pull Requests
+description: Checks if a PR is a fork and applies an external label accordingly
+
+inputs:
+  github_token:
+    description: The GitHub token to use read member status
+    required: true
+  team:
+    description: The team slug for external collaborators
+    required: true
+  org:
+    description: The organization to check membership in
+    required: true
+  username:
+    description: The username to check membership for
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/github-script@v7
+      id: set-result
+      with: 
+        github-token: ${{ inputs.github_token }}
+        script: |
+          const response = await github.rest.teams.getMembershipForUserInOrg({
+            org: "${{ inputs.org }}",
+            team_slug: "${{ inputs.team }}",
+            username: "${{ inputs.username }}",
+          });
+          return response.data.state === 'active';
+    - name: Add label
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ github.token }}
+      if: github.event.pull_request.head.repo.fork == true || steps.set-result.outputs.result == 'true'
+      run: gh pr edit ${{ github.event.pull_request.html_url }} --add-label "external"
```

Same suggestion as above

```suggestion
    - name: Add label
      shell: bash
      env:
        PR_HTML_URL: ${{ github.event.pull_request.html_url }}
        GITHUB_TOKEN: ${{ github.token }}
      if: github.event.pull_request.head.repo.fork == true || steps.set-result.outputs.result == 'true'
      run: gh pr edit "$PR_HTML_URL" --add-label "external"
```